datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/mlmodel/mappers/MlModelMapper.java

```diff
@@ -0,0 +1,96 @@
+package com.linkedin.datahub.graphql.types.mlmodel.mappers;
+
+import java.util.stream.Collectors;
+
+import javax.annotation.Nonnull;
+
+import com.linkedin.data.template.RecordTemplate;
+import com.linkedin.datahub.graphql.generated.EntityType;
+import com.linkedin.datahub.graphql.generated.FabricType;
+import com.linkedin.datahub.graphql.generated.MLModel;
+import com.linkedin.datahub.graphql.generated.SourceCode;
+import com.linkedin.datahub.graphql.types.common.mappers.CostMapper;
+import com.linkedin.datahub.graphql.types.common.mappers.DeprecationMapper;
+import com.linkedin.datahub.graphql.types.common.mappers.InstitutionalMemoryMapper;
+import com.linkedin.datahub.graphql.types.common.mappers.OwnershipMapper;
+import com.linkedin.datahub.graphql.types.common.mappers.StatusMapper;
+import com.linkedin.datahub.graphql.types.mappers.ModelMapper;
+
+/**
+ * Maps Pegasus {@link RecordTemplate} objects to objects conforming to the GQL schema.
+ *
+ */
+public class MlModelMapper implements ModelMapper<com.linkedin.ml.MLModel, MLModel> {
+
+    public static final MlModelMapper INSTANCE = new MlModelMapper();
+
+    public static MLModel map(@Nonnull final com.linkedin.ml.MLModel mlModel) {
+        return INSTANCE.apply(mlModel);
+    }
+
+    @Override
+    public MLModel apply(@Nonnull final com.linkedin.ml.MLModel mlModel) {
+        MLModel result = new MLModel();
+        result.setUrn(mlModel.getUrn().toString());
+        result.setType(EntityType.MLMODEL);
+        result.setName(mlModel.getName());
+        result.setDescription(mlModel.getDescription());
+        result.setOrigin(Enum.valueOf(FabricType.class, mlModel.getOrigin().name()));
+        result.setTags(mlModel.getTags());
+
+        if (mlModel.getOwnership() != null) {
+            result.setOwnership(OwnershipMapper.map(mlModel.getOwnership()));
+        }
+        if (mlModel.getMlModelProperties() != null) {
+            result.setMlModelProperties(MLModelPropertiesMapper.map(mlModel.getMlModelProperties()));
+        }
+        if (mlModel.getIntendedUse() != null) {
+            result.setIntendedUse(IntendedUseMapper.map(mlModel.getIntendedUse()));
+        }
+        if (mlModel.getMlModelFactorPrompts() != null) {
+            result.setMlModelFactorPrompts(MLModelFactorPromptsMapper.map(mlModel.getMlModelFactorPrompts()));
+        }
+        if (mlModel.getMetrics() != null) {
+            result.setMetrics(MetricsMapper.map(mlModel.getMetrics()));
+        }
+        if (mlModel.getEvaluationData() != null) {
+            result.setEvaluationData(mlModel.getEvaluationData().getEvaluationData().stream().map(BaseDataMapper::map).collect(Collectors.toList()));
```

Comment:
both getEvaluationData() and getEvaluationData().getEvaluationData() will not return null, right? Otherwise risk of NPE

Reply:
getEvaluationData().getEvaluationData() does not return null it is a mandatory field - https://github.com/linkedin/datahub/blob/master/metadata-models/src/main/pegasus/com/linkedin/ml/metadata/EvaluationData.pdl#L13